task-3/results/app/export_data.py

```
@@ -0,0 +1,259 @@
+            # Use custom query or default to full table
+            if query:
+                cursor.execute(query)
+            else:
+                # For daily export, get only yesterday's data where applicable
+                yesterday = (datetime.now() - timedelta(days=1)).date()
+
+                if table_name in ['shipments', 'shipment_events']:
+                    query = f"""
+                        SELECT * FROM {table_name}
+                        WHERE DATE(created_at) = '{yesterday}'
+                    """
```

Ruff (0.13.1)

Possible SQL injection vector through string-based query construction

(S608)